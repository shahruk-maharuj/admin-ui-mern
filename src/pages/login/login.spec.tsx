import { it, describe, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import Login from "./login";

describe("Login page", () => {
  it("should render with required fields", () => {
    render(<Login />);

    expect(screen.getByPlaceholderText("Username")).toBeInTheDocument();
    expect(screen.getByPlaceholderText("Password")).toBeInTheDocument();
    expect(
      screen.getByRole("button", { name: /sign in/i })
    ).toBeInTheDocument();
    expect(
      screen.getByRole("checkbox", { name: /remember me/i })
    ).toBeInTheDocument();
    expect(screen.getByText("Forgot password?")).toBeInTheDocument();
  });
});
