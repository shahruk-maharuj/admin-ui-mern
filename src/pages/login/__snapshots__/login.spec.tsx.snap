// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Login page > should render with required fields 1`] = `
<div>
  Login
</div>
`;
